llvm/lib/Target/NVPTX/NVPTXISelLowering.cpp

```diff
@@ -207,8 +209,13 @@ static void ComputePTXValueVTs(const TargetLowering &TLI, const DataLayout &DL,
       // Vectors with an even number of f16 elements will be passed to
       // us as an array of v2f16/v2bf16 elements. We must match this so we
       // stay in sync with Ins/Outs.
-      if ((Isf16Orbf16Type(EltVT.getSimpleVT())) && NumElts % 2 == 0) {
-        EltVT = EltVT == MVT::f16 ? MVT::v2f16 : MVT::v2bf16;
+      if ((Is16bitsType(EltVT.getSimpleVT())) && NumElts % 2 == 0) {
+        if (EltVT == MVT::f16)
```

Comment:
Nit: I'd use `switch(EltVT)`. This is fine, too.

Reply:
EltVT is not an enum so using it in a switch doesn't work out of the box. So I left it like that, let me know if you see a better way.

Reply:
`switch(EltVT.SimpleTy)` should work.

Reply:
Done